lib/src/firestore_cache.dart

```diff
@@ -77,19 +79,31 @@ class FirestoreCache {
   /// You can also pass in [localCacheKey] as the key for storing the last local
   /// cache date, and [isUpdateCacheDate] to set if it should update the last
   /// local cache date to current date and time.
-  static Future<QuerySnapshot<Map<String, dynamic>>> getDocuments({
-    required Query<Map<String, dynamic>> query,
-    required DocumentReference<Map<String, dynamic>> cacheDocRef,
+  ///
+  /// If you are using firestore collection withConverter then you have to also
+  /// pass toFirestore method to converter
+  static Future<QuerySnapshot<T>> getDocuments<T>({
+    required Query<T> query,
+    required DocumentReference<T> cacheDocRef,
```

Comment:
I can understand the use case for `Query<T>` then returning `QuerySnapshot<T>`. But can people simply pass in the default `Map<String, dynamic>` for `cacheDocRef`? This document is simply used for fetching the cache date info, so I would imagine there's no need to use the parameter with the converters.

Also, I don't think it can use the same type `T` which is used for `Query` and the returned `QuerySnapshot`>, as it probably has a different type,